### PR TITLE
[XPU]add squeeze2+matmul pass

### PR DESCRIPTION
**Pull request by wz1qqx**
### PR types

Performance optimization
### PR changes

OPs
### Description

- add squeeze2+matmul fuse pass
- add relu6 fuse pass

### CODE REVIEW

**Review by zhupengyang — DISMISSED** [review]
LGTM

后面的pass都尽量用python单测

**Comment by wz1qqx** [comment]
> LGTM
> 
> 后面的pass都尽量用python单测

好的

File: paddle/fluid/framework/ir/xpu/reshape2_matmul_xpu_fuse_pass.cc [hunk 1]
```diff
@@ -139,6 +139,76 @@ Reshape2MatmulPattern::Reshape2MatmulPattern(PDPattern* pattern,
...
+  auto* squeeze2_in =
+      pattern->NewNode(squeeze2_in_repr())
+          ->assert_is_op_input("squeeze2", "X")
+          ->AsInput()
+          ->assert_more([](Node* node) {
+            auto squeeze2_in_x_shape = node->Var()->GetShape();
+            size_t squeeze2_in_rank = squeeze2_in_x_shape.size();
+            bool nice_shape =
+                squeeze2_in_x_shape[2] == 1 && squeeze2_in_x_shape[3] == 1;
+            return squeeze2_in_rank == 4 && nice_shape;
```
Comment (zhupengyang) [inline paddle/fluid/framework/ir/xpu/reshape2_matmul_xpu_fuse_pass.cc:168]:
```suggestion
            return squeeze2_in_rank == 4 && (squeeze2_in_x_shape[2] == 1 && squeeze2_in_x_shape[3] == 1);
```

如果 ```squeeze2_in_rank != 4```，```squeeze2_in_x_shape[2] == 1 && squeeze2_in_x_shape[3] == 1``` 可能直接报错

**Review by zhupengyang — APPROVED** [review]

File: paddle/fluid/framework/ir/xpu/reshape2_matmul_xpu_fuse_pass.cc (hunk 1 above)
Comment (wz1qqx) [reply paddle/fluid/framework/ir/xpu/reshape2_matmul_xpu_fuse_pass.cc:168]: 嗯 确实存在这个隐患。。下个pr会将nice_shape的判断加到 if (squeeze2_in_rank == 4) 的判断中

**Pull request merged**